### DISCUSSION TITLE
lxqt own window manager.

### DISCUSSION THREAD

**Discussion by fadi123go** [lxqt/lxqt#2415] [Ideas]
I think you should start developing your own window manager. Openbox has not improved for more than a decade and it is a relic of the old days of lxde. You should not be always dependent on other desktop environment for a window manager,  most of which are gtk based . Some people do not like to have more than one desktop environment on their system. lxqt has come a long way to become a full independent desktop  environment but it stands among the rest as  the only without a window manager of its own . You should build your own qt-based, light, up-to-date , well integrated window manager, not some kwin fork .

**Comment by tsujan** [comment]
This was decided a long time ago. LXQt is and will remain WM-agnostic. We've also started to keep it so on Wayland, especially with `wlroots`-based Wayland compositors.

**Comment by AzumaHazuki** [comment]
> I think you should start developing your own window manager. Openbox has not improved for more than a decade and it is a relic of the old days of lxde. You should not be always dependent on other desktop environment for a window manager, most of which are gtk based . Some people do not like to have more than one desktop environment on their system. lxqt has come a long way to become a full independent desktop environment but it stands among the rest as the only without a window manager of its own . You should build your own qt-based, light, up-to-date , well integrated window manager, not some kwin fork . You should not slack off on the idea . Try hard and you shall be rewarded.

What Tsujan said...also, LXQt is meant to be lightweight and *modular,* meaning it should be fairly easy to swap parts out and we want as little breakage as possible. Openbox is old, yes, but it's also rock-stable. If the writer of LabWC were committed to making sure all the theming primitives that worked in Openbox worked there too, we'd have a drop-in replacement to make X and Wayland sessions utterly seamless.

**Reply by tsujan** [reply]
> Openbox is old, yes, but it's also rock-stable

Moreover, modularity means that if someone makes a good and bug-free X11 WM by forking Openbox (which is quite possible, even if not done yet), it could be used under LXQt too. But that's not the job of LXQt.

**Comment by fadi123go** [comment]
Never asked to break the modularity of lxqt .  I use xfwm4 , which means lxqt can not exist without xfce. Creating a modern window manager module should always be within the scope of any desktop environement . Anyway great job on what you have done with lxqt so far. Peace out.

**Reply by ghost** [reply]
> Never asked to break the modularity of lxqt . I use xfwm4 , which means lxqt can not exist without xfce. Creating a modern window manager module should always be within the scope of any desktop environement . Anyway great job on what you have done with lxqt so far. Peace out.

With Wayland, WMs are a thing of past now its called compositors, you could actually with some work write your own Mir aka Wayland compositor it isnt that difficult for simple uses but you can surely do extra code for more functionality.

**Reply by stefonarch** [reply]
No need to reinvent the wheel,  many existing compositors can [already be used](https://github.com/stefonarch/LXQt-Wayland-files) with (most) lxqt-components.

**Reply by tsujan** [reply]
> With Wayland, WMs are a thing of past now

In the case of Wayland, the WM and compositor always come together because Wayland is always composited. Window management is an important job that can never be "a thing of past".

> you could actually with some work write your own Mir aka Wayland....

Already discussed [elsewhere](https://github.com/lxqt/lxqt/issues/10), for years.

**Reply by ghost** [reply]
> > With Wayland, WMs are a thing of past now
> 
> In the case of Wayland, the WM and compositor always come together because Wayland is always composited. Window management is an important job that can never be "a thing of past".
> 
> > you could actually with some work write your own Mir aka Wayland....
> 
> Already discussed [elsewhere](https://github.com/lxqt/lxqt/issues/10), for years.

What i actually meant to say was now Window managers are not called WMs individually instead now they are called compositors....

**Comment by fadi123go** [comment]
Sorry to reignite this topic again but I can see the transition to Wayland is going very well. And I just want to re-express my wish to see a window manger module for lxqt. It should be qt based, light and basic with simple compositing effects of its own like shadows and transparency . And you can add more feature to it throughout the years to come. I hate to see Lxqt always depended on external window managers.

**Reply by stefonarch** [reply]
If you encounter 3 or 4 wayland-skilled developers with spare time you can send them here to work for some years... 

I see it positive: having so many choices now on Wayland available.

**Reply by tsujan** [reply]
> ... a window manger module for lxqt...

I'm glad that we never made that mistake. The value of LXQt's decision to remain WM-agnostic is especially evident on Wayland.
